internal/worker/fortress/fortress.go

```diff
@@ -159,6 +159,7 @@ func (ticket guestTicket) complete(finished func()) {
 	select {
 	case <-ticket.ctx.Done():
 		ticket.result <- ErrAborted
-	case ticket.result <- ticket.visit():
+	default:
+		ticket.result <- ticket.visit()
```

Comment:
Can we guarantee someone is reading on the other end, otherwise this will also block forever, because it's not guarded by an exit clause.

If there is someone reading, then this become moot.

Reply:
Yeah, someone is always reading this channel. This change just ensure the visit result is always sent, rather than sometimes it being never sent but visit still run.
```go
// Visit is part of the Guest interface.
func (f *fortress) Visit(ctx context.Context, visit Visit) error {
	result := make(chan error)
	select {
	case <-f.tomb.Dying():
		return ErrShutdown
	case <-ctx.Done():
		return ErrAborted
	case f.guestTickets <- guestTicket{
		ctx:    ctx,
		visit:  visit,
		result: result,
	}:
		return <-result
	}
}
```

Reply:
This change attempts to deal with this https://go.dev/play/p/Z8EEoaZ850k

RHS evaluation is done before the select on available channels. So they should not be expensive or have side-effects.